blues/moves.py

```diff
@@ -177,70 +178,304 @@ def move(self, context):
-        on a context.
-        Parameters
-        ----------
-        context : openmm.context object
-        OpenMM context whose positions should be moved.
-        """
-        rand_num = np.random.choice(len(self.probs), p=self.probs)
-        try:
-            new_context = self.moves[rand_num].move(context)
-        except Exception as e:
-            #In case the move isn't properly implemented, print out useful info
-            print('Error: move not implemented correctly, printing traceback:')
-            ex_type, ex, tb = sys.exc_info()
-            traceback.print_tb(tb)
-            print(e)
-            raise SystemExit
-
-        return new_context
+        atom_indices = self.atom_indices
+        context = nc_context
+
+        stateinfo = context.getState(True, True, False, True, True, False)
+        oldDartPos = stateinfo.getPositions(asNumpy=True)
+        lig_pos = np.asarray(oldDartPos._value)[self.atom_indices]*unit.nanometers
+        self._findDart(context)
+        center = self.getCenterOfMass(lig_pos, self.masses)
+        selectedboard, changevec = self._calc_from_center(com=center)
+        if selectedboard != None:
+            #TODO just use oldDartPos instead of using temp newDartPos
+            newDartPos = np.copy(oldDartPos)
+            comMove = self._reDart(changevec)
+            vecMove = comMove - center
+            for residue in atom_indices:
+                newDartPos[residue] = newDartPos[residue] + vecMove._value
+            context.setPositions(newDartPos)
+
+            return context
+
+    def _calc_from_center(self, com):
+
+        distList = []
+        diffList = []
+        indexList = []
+        #Find the distances of the COM to each dart, appending
+        #the results to distList
+        for dart in self.dartboard:
+            diff = com - dart
+            dist = np.sqrt(np.sum((diff)*(diff)))*unit.nanometers
+            distList.append(dist)
+            diffList.append(diff)
+        selected = []
+        #Find the dart(s) less than self.dart_size
+        for index, entry in enumerate(distList):
+            if entry <= self.dart_size:
+                selected.append(entry)
+                diff = diffList[index]
+                indexList.append(index)
+        #Dart error checking
+        #to ensure reversibility the COM should only be
+        #within self.dart_size of one dart
+        if len(selected) == 1:
+            return selected[0], diffList[indexList[0]]
+        elif len(selected) == 0:
+            return None, diff
+        elif len(selected) >= 2:
+            #COM should never be within two different darts
+            raise ValueError('sphere size overlap, check darts')
+            #TODO can treat cases using appropriate probablility correction
+
+    def _findDart(self, nc_context):
+        """
+        Helper function to dynamically update dart positions based on positions
+        of other particles.
```

Comment:
Could be slightly more clear -- positions of which other particles? (If I wanted to read the doc string to understand what it does, as in fact I do as part of my code review, this doesn't help me that much.) I assume what you're trying to do here is to use the rest of the protein to figure out where the region you're going to dart to has moved to in the interim??